npm/src/controller/oauth.ts

```diff
@@ -929,36 +942,40 @@ export class OAuthController implements IOAuthController {
       }
 
       await this.sessionStore.delete(RelayState);
-
       return { redirect_url: redirect.success(redirect_uri!, params) };
     } catch (err: any) {
       const { error, error_description, error_uri, session_state, scope, stack } = err;
       const error_message = error_description || getErrorMessage(err);
-      const traceId = await this.ssoTraces.saveTrace({
-        error: error_message,
-        context: {
-          tenant: oidcConnection.tenant,
-          product: oidcConnection.product,
-          clientID: oidcConnection.clientID,
-          providerName: oidcConnection.oidcProvider.provider,
-          redirectUri: redirect_uri,
-          relayState: RelayState,
-          isSAMLFederated,
-          isOIDCFederated,
-          acsUrl: session.requested.acsUrl,
-          entityId: session.requested.entityId,
-          requestedOIDCFlow: !!session.requested.oidc,
-          oidcIdPRequest: session?.requested?.oidcIdPRequest,
-          profile,
-          error,
-          error_description,
-          error_uri,
-          session_state_from_op_error: session_state,
-          scope_from_op_error: scope,
-          stack,
-          oidcTokenSet: { id_token: tokens?.id_token, access_token: tokens?.access_token },
-        },
-      });
+      let traceId: string | undefined;
+
+      if (!shouldDisableSsoTrace) {
+        traceId = await this.ssoTraces.saveTrace({
```

Comment:
Isn't it better to do this check in ssoTraces.saveTrace instead of all places that call it?